qiskit/transpiler/__init__.py

```diff
@@ -19,8 +19,403 @@
+
+      ghz = QuantumCircuit(5)
+      ghz.h(0)
+      ghz.cx(0,range(1,5))
+      ghz.draw(output='mpl')
+
+
+   .. jupyter-execute::
+
+      depths = []
+      for _ in range(100):
+         depths.append(transpile(ghz,
+                                 backend,
+                                 initial_layout=[7, 0, 4, 15, 19],
+                                ).depth())
+
+      plt.figure(figsize=(8, 6))
+      plt.hist(depths, bins=list(range(14,36)), align='left', color='#AC557C')
+      plt.xlabel('Depth', fontsize=14)
+      plt.ylabel('Counts', fontsize=14);
+
+
+   This distribution is quite wide, signaling the difficultly the SWAP mapper is having
+   in computing the best mapping.  Most circuits will have a distribution of depths,
+   perhaps not as wide as this one, due to the stochastic nature of the default SWAP
+   mapper.  Of course, we want the best circuit we can get, especially in cases where
+   the depth is critical to success or failure.  In cases like this, it is best to
+   :func:`transpile` a circuit several times, e.g. 10, and take the one with the
+   lowest depth.  The :func:`transpile` function will automatically run in parallel
+   mode, making this procedure relatively speedy in most cases.
+
+   .. raw:: html
+
+      <br>
+
+
+.. container:: toggle
+
+   .. container:: header
+
+      **Gate Optimization**
+
+   Decomposing quantum circuits into the basis gate set of the IBM Quantum devices,
+   and the addition of SWAP gates needed to match hardware topology, conspire to
+   increase the depth and gate count of quantum circuits.  Fortunately many routines
+   for optimizing circuits by combining or eliminating gates exist.  In some cases
+   these methods are so effective the output circuits have lower depth than the inputs.
+   In other cases, not much can be done, and the computation may be difficult to
+   perform on noisy devices.  Different gate optimizations are turned on with
+   different `optimization_level` values.  Below we show the benefits gained from
+   setting the optimization level higher:
+
+   .. important::
+
+      The output from :func:`transpile` varies due to the stochastic swap mapper.
+      So the numbers below will likely change each time you run the code.
+
+
+   .. jupyter-execute::
+
+      import matplotlib.pyplot as plt
+      from qiskit import QuantumCircuit, transpile
+      from qiskit.test.mock import FakeBoeblingen
+      backend = FakeBoeblingen()
+
+      ghz = QuantumCircuit(5)
+      ghz.h(0)
+      ghz.cx(0,range(1,5))
+      ghz.draw(output='mpl')
+
+
+   .. jupyter-execute::
+
+      for kk in range(4):
+         circ = transpile(ghz, backend, optimization_level=kk)
+         print('Optimization Level {}'.format(kk))
+         print('Depth:', circ.depth())
+         print('Gate counts:', circ.count_ops())
+         print()
+
+
+   .. raw:: html
+
+      <br>
+
+
+Transpiler API
+==============
+
 Pass Management
-===============
+---------------
```

Comment:
Can we add `transpiler.preset_passmanagers` to the list of documented modules? There is helpful information in there about what exactly each level does, but right now it is not picked up by sphinx.

Reply:
Sure.

Reply:
This is already at the top-level of the documentation.